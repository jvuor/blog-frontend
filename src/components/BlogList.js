import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { Item, Icon } from 'semantic-ui-react'
import Moment from 'moment'
import stripMarkdown from '../utils/stripMarkdown'
import titleLink from '../utils/titleLink'
import truncateContent from '../utils/truncateContent'

export class BlogList extends React.Component {
  render () {
    return (
      <Item.Group>
        {this.props.blogs.map(blog => 
          <Item key={blog.id}>
            <Icon name='envelope open outline' />
            <Item.Content>
              <Link to={titleLink(blog.title, blog.id)}>
                <Item.Header>{blog.title}</Item.Header>
              </Link>
              <Item.Description>
                {stripMarkdown(truncateContent(blog.content))}
              </Item.Description>
              <Item.Extra>
                Written by {blog.user.name} {Moment(blog.created).fromNow()}
              </Item.Extra>
            </Item.Content>
          </Item>
        )}
      </Item.Group>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    blogs: state.blogs
  }
}

export default connect(mapStateToProps, null)(BlogList)