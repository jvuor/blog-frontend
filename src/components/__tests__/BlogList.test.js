import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { BlogList } from '../BlogList'
import { shallow } from 'enzyme'
import blogTestData from './blogTestData'

describe('BlogList component unit tests', () => {
  test('fails on empty redux state', () => {
    const blogs = null
    expect(() => shallow(<BlogList blogs={blogs} />)).toThrow()
  })

  test('displays all the correct data when given a valid state', () => {
    const wrapper = shallow(
      <MemoryRouter>
        <BlogList blogs={blogTestData} />
      </MemoryRouter>
    )
    const content = wrapper.html()

    blogTestData.forEach(blog => {
      expect(content).toContain(blog.title)
      expect(content).toContain(blog.user.name)
    })

    expect(content).toContain('<a href="/5b58dce799f2274cc85b0933/Markdown-Test">')
  })

  test('snapshot test', () => {
    const wrapper = shallow(
      <MemoryRouter>
        <BlogList blogs={blogTestData} />
      </MemoryRouter>
    )
    const content = wrapper.html()
    expect(content).toMatchSnapshot()
  })
})