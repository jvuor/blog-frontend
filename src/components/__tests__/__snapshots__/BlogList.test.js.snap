// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogList component unit tests snapshot test 1`] = `
"<div class=\\"ui items\\"><div class=\\"item\\"><i aria-hidden=\\"true\\" class=\\"envelope open outline icon\\"></i><div class=\\"content\\"><a href=\\"/5b58dce799f2274cc85b0933/Markdown-Test\\"><div class=\\"header\\">Markdown Test</div></a><div class=\\"description\\"> Subheader

Hi, this text has markdown! Hell yeah

link

 List Item
 Another Item

 Another Subheader

Still marking down like a boss yeaaaaaahhhh</div><div class=\\"extra\\">Written by Test User 3 days ago</div></div></div><div class=\\"item\\"><i aria-hidden=\\"true\\" class=\\"envelope open outline icon\\"></i><div class=\\"content\\"><a href=\\"/5b5895867296874dacbc9d6d/Test-Post-From-Admin-Panel\\"><div class=\\"header\\">Test Post From Admin Panel</div></a><div class=\\"description\\">content here</div><div class=\\"extra\\">Written by Test User 3 days ago</div></div></div><div class=\\"item\\"><i aria-hidden=\\"true\\" class=\\"envelope open outline icon\\"></i><div class=\\"content\\"><a href=\\"/5b55dd6774d88b36481f5607/Second-Test-Post\\"><div class=\\"header\\">Second Test Post</div></a><div class=\\"description\\">Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt (...)</div><div class=\\"extra\\">Written by Test User 5 days ago</div></div></div><div class=\\"item\\"><i aria-hidden=\\"true\\" class=\\"envelope open outline icon\\"></i><div class=\\"content\\"><a href=\\"/5b55dcdb74d88b36481f5606/Test-post\\"><div class=\\"header\\">Test post</div></a><div class=\\"description\\">Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt (...)</div><div class=\\"extra\\">Written by Test User 5 days ago</div></div></div></div>"
`;
